apigw-sqs-lambda-sns/webhook_sns/webhook_sns_stack.py

```diff
@@ -0,0 +1,149 @@
+from aws_cdk import (
+    Stack,
+    aws_apigateway as apigateway,
+    aws_sqs as sqs,
+    aws_lambda as _lambda,
+    aws_iam as iam,
+    aws_lambda_event_sources as lambda_event_sources,
+    CfnOutput,
+    Duration
+)
+from constructs import Construct
+
+
+class WebhookSnsStack(Stack):
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Create SQS Queue
+        queue = sqs.Queue(
+            self, "WebhookQueue",
+            queue_name="webhook-queue"
+        )
+
+        # Create Lambda function
+        lambda_function = _lambda.Function(
+            self, "EventProcessingFunction",
+            runtime=_lambda.Runtime.PYTHON_3_9,
+            handler="app.lambda_handler",
+            code=_lambda.Code.from_asset("lambda"),
+            timeout=Duration.seconds(30)
+        )
+
+        # Grant Lambda permission to publish to SNS
+        lambda_function.add_to_role_policy(
+            iam.PolicyStatement(
+                effect=iam.Effect.ALLOW,
+                actions=["sns:Publish"],
+                resources=["*"]
```

Comment:
Lambda function has permission to publish to all SNS resources using wildcard "*" which violates least privilege principle.

Restrict SNS publish permissions to specific topic ARN or use account-level resource constraint.

```
...
resources=[f"arn:aws:sns:{self.region}:{self.account}:*"]
...
```